### ISSUE TITLE
cli: fix release-helm tag logic

### ISSUE THREAD

**Issue by diegodelemos** [reanahub/reana#497]
Currently, almost all `maint-0.7` releases are incorrectly tagged. This happens when following the [Releasing docs](https://github.com/reanahub/reana/wiki/Releasing) tagging a maintenance branch. Tags end up pointing to the latest `master` instead of the correct latest commit in `maint-x.y`.

All Helm releases are fortunately correct so no harm has been done on that side (the `.tar.gz` contain the correct versions).

The actions to be taken are:
- [x] Fix all past release tags to point to the correct release commits. Note this involves force pushing, so we need to warn people.
- [x] Fix the `release-helm` command itself to not behave like this.

**Comment by mvidalgarcia** [comment]
This corresponds (or at least partially) to the bullet point "A: fix bad 0.7.x tags (due to cr)" from our December 2021 checkpoint.

**Comment by mvidalgarcia** [comment]
Tags fixed:

```console
$ git tag -ln 
0.7.0           release: 0.7.0
0.7.0-alpha.1   release: 0.7.0-alpha.1
0.7.0-alpha.2   release: 0.7.0-alpha.2
0.7.1           release: 0.7.1
0.7.2           release: 0.7.2
0.7.3           release: 0.7.3
0.7.4           release: 0.7.4
0.7.4-alpha.1   release: 0.7.4-alpha.1
0.7.4-alpha.2   release: 0.7.4-alpha.2
0.8.0           release: 0.8.0
0.8.0-alpha.1   release: 0.8.0a1
0.8.0-alpha.2   release: 0.8.0-alpha.2
0.8.0-alpha.3   release: 0.8.0-alpha.3
0.8.0-alpha.4   release: 0.8.0-alpha.4
0.8.0-alpha.5   release: 0.8.0-alpha.5
v0.1.0          REANA v0.1.0
v0.2.0          REANA v0.2.0
v0.3.0          REANA v0.3.0
v0.4.0          REANA v0.4.0
v0.5.0          REANA v0.5.0
v0.6.0          REANA v0.6.0
v0.6.1          REANA v0.6.1
```
